connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/lifecycle/InboundConnectorManager.java

```diff
@@ -167,7 +167,19 @@ private void activateConnector(InboundConnectorDefinitionImpl newConnector) {
       if (executable instanceof PollingConnectorExecutable) {
         Long processDefinitionKey = newConnector.processDefinitionKey();
         Supplier<List<ProcessInstance>> processInstancesSupplier =
-            () -> processDefinitionSearch.fetchProcessInstancesWithVariables(processDefinitionKey);
+            () ->
+                processDefinitionSearch
+                    .fetchProcessInstancesByDefinitionKey(processDefinitionKey)
+                    .stream()
+                    .map(
+                        instance ->
+                            new ProcessInstance(
+                                instance.getKey(),
+                                () ->
+                                    processDefinitionSearch.fetchVariablesByProcessInstanceKey(
+                                        instance.getKey())))
+                    .collect(Collectors.toList());
+
         executable.activate(
             new InboundIntermediateConnectorContextImpl(inboundContext, processInstancesSupplier));
```

Comment:
The `interface` hierarchy should be sealed and the `PollingConnectorExecutable` should receive the actual interface: `InboundIntermediateConnectorContext` without a need to cast it. Please move the polling parts into a subpackage `polling` similar to the `webhook` connector.

Reply:
The `PollingConnectorExecutable` and the `WebhookConnector` executable should be sealed interfaces of the `InboundConnectorExecutable`.